File: Contracts/Model/Product/Response/ProductResponse.cs
namespace Contracts.Model.Product.Response;

public class ProductResponse : ByBaseResponse
{
	public Guid Id { get; set; }
	public string Code { get; set; }
	public bool IsService { get; set; }
	public OptionsResponse Option { get; set; }
	public CategoriesListResponse Category { get; set; }
	public SuppliersListResponse? SupplierId { get; set; }
	public WarrantiesListResponse? Warranty { get; set; }
	public ICollection<OptionsResponse> Options { get; set; }
}

File: ClientApp/src/app/core/interfaces/Product/Response/productResponse.ts
import {ByBaseResponse} from "../../byBaseResponse";
import {OptionsResponse} from "./optionsResponse";
import {SuppliersListResponse} from "./suppliersListResponse";
import {WarrantiesListResponse} from "./warrantiesListResponse";
import {CategoriesListResponse} from "@core/interfaces/Product/category";

export interface ProductResponse extends ByBaseResponse {
  id: string;
  code: string;
  isService: boolean;
  option: OptionsResponse;
  category: CategoriesListResponse;
  warranty: WarrantiesListResponse | null;
  supplier: SuppliersListResponse | null;
  options: OptionsResponse[];
}
